lib/net/wifi.toit

```diff
@@ -13,8 +13,56 @@ CONFIG_PASSWORD  /string ::= "wifi.password"
 CONFIG_BROADCAST /string ::= "wifi.broadcast"
 CONFIG_CHANNEL   /string ::= "wifi.channel"
 
+CONFIG_SCAN_CHANNELS /string ::= "scan.channels"
+CONFIG_SCAN_PASSIVE  /string ::= "scan.passive"
+CONFIG_SCAN_PERIOD   /string ::= "scan.period"
+
+SCAN_AP_SSID     /string ::= "scan.ap.ssid"
+SCAN_AP_BSSID    /string ::= "scan.ap.bssid"
+SCAN_AP_RSSI     /string ::= "scan.ap.rssi"
+SCAN_AP_AUTHMODE /string ::= "scan.ap.authmode"
+SCAN_AP_CHANNEL  /string ::= "scan.ap.channel"
+
+WIFI_SCAN_SSID_     ::= 0
+WIFI_SCAN_BSSID_    ::= 1
+WIFI_SCAN_RSSI_     ::= 2
+WIFI_SCAN_AUTHMODE_ ::= 3
+WIFI_SCAN_CHANNEL_  ::= 4
+WIFI_SCAN_ELEMENT_COUNT_ ::= 5
+
+SCAN_TIMEOUT_MS_/int := 1000 /* microseconds */ 
+
 service_/WifiServiceClient? ::= (WifiServiceClient --no-open).open
 
+class AccessPoint:
+  ssid/string := ?
+  bssid/ByteArray := ?
+  rssi/int := ?
+  authmode/int := ?
+  channel/int := ?
+
+  static WIFI_AUTHMODE_NAME_/Map ::= {
+    0: "Open",
+    1: "WEP",
+    2: "WPA PSK",
+    3: "WPA2 PSK",
+    4: "WPA/WPA2 PSK",
+    5: "WPA2 Enterprise",
+    6: "WPA3 PSK",
+    7: "WPA2/WPA3 PSK",
+    8: "WAPI PSK",
+  }
+
+  constructor .ssid/string .bssid/ByteArray .rssi/int .authmode/int .channel/int:
```

Comment:
It might be nice to use named arguments for this (and the types are inferred from the fields):

```suggestion
  constructor --.ssid --.bssid --.rssi --.authmode --.channel:
```